src/event-manager/event-channel-client.ts

```diff
@@ -121,9 +143,22 @@ export default class EventChannelClient {
     };
   }
 
+  sendMessage(message: EventChannelMessage) {
+    if (!this._isConnected) {
+      throw Error(`Failed to send message. Event channel is not connected!`);
```

Comment:
Would it be better to just log the message and return? For example, when unsubscribe() is called and somehow it's disconnected, it will throw an error and deleteFilter() will not get executed.

Reply:
Since `sendMessage` function is using in many places., how about change the order? `deleteFilter` can also throw an error, but that's a really exceptional case, so I guess that's fine.
```
    try {
      const filter = this._eventCallbackManager.getFilter(filterId);
      this._eventCallbackManager.deleteFilter(filter.id);
      this._eventChannelClient.deregisterFilter(filter);
      callback(null, filter);
    } catch (err) {
      callback(err, null);
    }
```

Reply:
in another way, before calling this._eventChannelClient.deregisterFilter, it would be good to check whether it is connected or not and execute it only if it is connected. In the case of disconnected state, as you said, simply print with console.log?

Reply:
discussed offline) Let's add the connection checks AND change the error throwing to error logging

Reply:
👍 Done! PTAL @liayoo

Reply:
Looks good, thanks!